crates/uv/src/commands/tool/run.rs

```diff
@@ -104,6 +105,37 @@ pub(crate) async fn run(
         return Err(anyhow::anyhow!("Tool command could not be parsed as UTF-8 string. Use `--from` to specify the package name."));
     };
 
+    if let Some(ref f) = from {
+        if Path::new(&f)
+            .extension()
+            .is_some_and(|ext| ext.eq_ignore_ascii_case("py") || ext.eq_ignore_ascii_case("pyw"))
+        {
+            return Err(anyhow::anyhow!(
+                "It looks you have passed a script instead of a package into `--from`"
+            ));
+        }
+    } else {
+        let possible_script = Path::new(target);
+        if possible_script
+            .extension()
+            .is_some_and(|ext| ext.eq_ignore_ascii_case("py") || ext.eq_ignore_ascii_case("pyw"))
+        {
+            return if possible_script.try_exists()? {
+                Err(anyhow::anyhow!(
+                    "It looks like you are trying to run a script. Did you mean `{}`?",
+                    format!("uv run {}", possible_script.display()).cyan()
+                ))
+            } else {
+                Err(anyhow::anyhow!(
+                    "It looks like you are trying to run a script that doesn't exist. \n\n{}{} Did you mean to run a tool with `uvx {}`?",
```

Comment:
Sorry I missed this initially, you need to use `invocation_source` here instead of hard-coding `uvx` because that's an alias for `uv tool run` (and we want to match what the user used)

Reply:
Updated